src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -141,73 +133,23 @@ internal sealed class MeterProviderSdk : MeterProvider
                         metricStreamConfigs.Add(null);
                     }
 
-                    var maxCountMetricsToBeCreated = metricStreamConfigs.Count;
-
-                    // Create list with initial capacity as the max metric count.
-                    // Due to duplicate/max limit, we may not end up using them
-                    // all, and that memory is wasted until Meter disposed.
-                    // TODO: Revisit to see if we need to do metrics.TrimExcess()
-                    var metrics = new List<Metric>(maxCountMetricsToBeCreated);
-                    lock (this.instrumentCreationLock)
+                    if (this.reader != null)
```

Comment:
Is `reader == null` a valid case? Thinking maybe ctor should throw in that case so we don't need to check it.

Reply:
Maybe it's fine. Similar discussion here https://github.com/open-telemetry/opentelemetry-dotnet/pull/2476.